test/test_payu.py

```diff
@@ -148,8 +148,19 @@ def test_read_config():
 
     assert(config.pop('collate') == {})
     assert(config.pop('control_path') == os.getcwd())
+    assert(config.pop('modules') == {})
     assert(config == {})
 
+    # Test transform legacy modules option 
+    with open(config_tmp, 'w') as f:
+        f.write('modules:\n   - test_user_defined_module')
+    
+    config = payu.fsops.read_config(config_tmp)
+    modules_config = config.get('modules', {})
+    
+    assert(modules_config.get('load', []) == ['test_user_defined_module'])
+    assert(modules_config.get('use', []) == [])
+
```

Comment:
It would be good to have some tests for `modules` in both the new and old syntax, and testing `use` as well.

It might be consistent to move the config file to the resources subdirectory you created?

Reply:
Ok, I've added a couple of tests and moved that config file. Generally is there a preference for creating/removing temporary test files when running the tests or using test files in the resources subdirectory?

Reply:
> Generally is there a preference for creating/removing temporary test files when running the tests or using test files in the resources subdirectory?

I'd say it depends. If there was a large number of test files based on, say, combinations of test variables, then it would make sense to generate them dynamically in a `setup` routine, and then remove them in `teardown`. In this case there aren't many files, and they're not large, and there is something appealing simple about being able to discover the file just by browsing the repo. Yes if it was encoded as a `dict` and written to a file it would be visible in the source code, but would be a little trickier to discover.

In the end there is a level of user preference as well.